doc/doc-style-rules.md

```diff
@@ -0,0 +1,134 @@
+# Documentation style rules
+
+Thank you for helping us improve the documentation! Below we have described some rules that we
+advise you to follow when you expand the documentation with new articles or add new sections to it.
+
+Any comments should be written in the following format:
+
+```
+> **Note**
+> Text of
+> note
+```
+
+The same goes for Warning:
+
+```
+> **Warning**
+> Text of
+> your warning
+```
+
+Do not use `->`, `=>` instead of `–`:
+
+```
+`@FN` – name of the current V function
+```
+
+All filenames must be **bold**:
+
+```
+**file.v**
+```
+
+All symbols (modules, functions, structures, etc.) must be `monospaced`:
+
+```
+Module `http.net`
+```
+
+A single terminal command may be written inline, but must be `monospaced`:
+
+```
+Command `v install`
+```
+
+Several terminal commands must be wrapped in \`\`\`:
+
+````
+```
+cd project
+v install markdown
+```
+````
+
+The language name must be capitalized:
+
+```
+V
+```
+
+Every sentence must end with a punctuation mark:
+
+```
+Please see the Troubleshooting section on our wiki page.
+```
+
+Use — or – instead of - when grammar requires it:
+
+```
+V needs some packages preinstalled — a working C compiler
+```
+
+Compiler names must be spelled correctly:
+
+```
+Clang, GCC, MSVC
+```
+
+Operating system names must be spelled correctly:
+
+```
+macOS, Linux, Windows
+```
```

Comment:
I do not see much value in requiring that. macos for example started as `Mac OS`, and is now `macOS`, and in another year can be `MaCOS`, or something else entirely. Consistency is good, but pandering to the whims of companies and their marketing departments is a never ending battle. 

The Linux and Windows names at least follow the common scheme of first Capital letter, and then the rest are lower cased. Imho if there should be requirement for the names, it should be that.

Reply:
I agree with @i582 . One change doesn't mean Apple will rename its operating system every year. They just came to consistency: iOS, macOS, and watchOS. Petr suggests coming to consistency too. Don't write `mac OS`, `Mac OS` or `maCOS`, as you said, but only `macOS`, which is the only and true way to name Apple's operating system. Don't pander to the whims of companies and their marketing department, just choose one name.

Reply:
All right, why not Macos, for consistency with Linux and Windows?

Reply:
> One change doesn't mean Apple will rename its operating system every year.

They do not do it every year, but they have done it more than once, with imho no guarantee they will not do it again (to say AppleOS):
https://en.wikipedia.org/wiki/MacOS_version_history

Reply:
Any official name change is one replacement command in one doc file. I don't see problems.

Reply:
This is a style guide, and it will serve as guidance for writing other documents, and can have ripple effects on all of them. In my opinion, this means that the rules in it are better to be immutable, or not present.

Reply:
In my opinion, the rules for the OS names, and for the compiler names imho are not needed, since in most cases, what is meant, is the actual names, used in the code (which are always lower cased), and rarely the official names of the corresponding projects.

Besides if you search for `MACOS`, `MacOS`, `macOS`, `macos` in the repo, you will see that all of these are used not only in V documents, but in the thirdparty/ folder too, where renaming them will lead to harder maintenance, when the corresponding library has a new upstream version. The same is valid for `clang` vs `Clang`, and for `gcc` vs `GCC`.

The cleanest solution for all those, is to just remove the rules for naming them from the document in this PR, and leave the naming decision back to where it belongs - to the individual developer that is making the changes, not to a blind rule that requires something without context.

Reply:
> to the individual developer that is making the changes,

This style guide was written only for documentation. I don't understand this problem. Why we should leave the naming decision back to where it belongs if belongs is the documentation?

Reply:
Here is an example: `thirdparty/mbedtls/README.md` .
That file contains both `GCC`, `Clang` as well as `clang` and `gcc`. What will you do with it?
Note, that we have not written any of it (it is part of `mbedtls`, which is a thirdparty project).

Reply:
I don't think this discussion has meaning because it will lead nowhere, and we'll waste time. You can make it as you see fit. I'm out of the discussion. Thank you.

Reply:
> Why we should leave the naming decision back to where it belongs if belongs is the documentation?

Because it is not necessary to require such spellings - developers are not stupid, and can think what is needed where, and even in the places where they may make mistakes, they are not so important that a whole rule should be targeted specifically to fix those.

Shall that guide also have a rule for "There shall be no spelling errors and typos in the documentation." ?

Reply:
> Here is an example: `thirdparty/mbedtls/README.md`.

`thirdparty/mbedtls/README.md` is not a part of V documentation.

Reply:
> `thirdparty/mbedtls/README.md` is not a part of V documentation.

How will you explain that to readers of the document in this PR?
Should it have a rule for what .md files it applies to as well, and/or known exceptions too?

Reply:
imho, the less bureaucracy, the better, which leads to my position on insisting on removing some of the rules here, instead of keeping them, then having to explain in more details where it does apply and how.